pubsub/subscriptions/main.go

```diff
@@ -104,25 +105,24 @@ func pullMsgs(client *pubsub.Client, name string, topic *pubsub.Topic) error {
 	}
 
 	// [START pull_messages]
-	sub := client.Subscription(name)
-	it, err := sub.Pull(ctx)
-	if err != nil {
-		return err
-	}
-	defer it.Stop()
-
 	// Consume 10 messages.
-	for i := 0; i < 10; i++ {
-		msg, err := it.Next()
-		if err == iterator.Done {
-			break
-		}
-		if err != nil {
-			return err
+	var mu sync.Mutex
+	received := 0
+	sub := client.Subscription(name)
+	receiveCtx, cancel := context.WithCancel(ctx)
+	sub.Receive(receiveCtx, func(ctx context.Context, msg *pubsub.Message) {
```

Comment:
You should still check for error. It could be that the service failed, and you'd want to know.